src/appleseed/foundation/image/genericimagefilewriter.cpp

```diff
@@ -60,15 +59,18 @@ void GenericImageFileWriter::write(
     const bf::path filepath(filename);
     const string extension = lower_case(filepath.extension().string());
 
-    if (extension == ".exr")
+    if (extension == ".exr" || extension == ".png")
     {
-        EXRImageFileWriter writer;
-        writer.write(filename, image, image_attributes);
-    }
-    else if (extension == ".png")
-    {
-        PNGImageFileWriter writer;
-        writer.write(filename, image, image_attributes);
+        OIIOImageFileWriter writer{ filename };
```

Comment:
Uniform initialization is not supported by VS 2012 which we must continue to support (fix everywhere).